src/socket.cpp

```diff
@@ -71,9 +109,10 @@ static inline bool tryReadSocket(int socket, void* data, size_t size, unsigned l
                 }
                 continue;
             }
-            throw ReadSocketException(SOCKET_LAST_ERRCODE, SOCKET_LAST_ERROR);
+#endif
+            throw std::runtime_error("Error reading from socket");
         } else if (r == 0) {
-            throw ReadSocketException(0, "Socket closed");
+            throw std::runtime_error("Socket closed");
```

Comment:
Why you drop `ReadSocketException`?

Reply:
Probably was just by accident, will fix it in the refactoring tonight